Src/main.c

```diff
@@ -882,11 +919,189 @@ getline(char **lineptr, size_t *n, FILE *stream)
 }
 #endif
 
+
+struct cli_args {
+  int T;
+  int usedopts;
+  int parse_run; // Set when parsing the options to -run
+};
+
+static char* prepend_arg(const char c, const char* arg) {
+  char* tmp = emalloc(strlen(arg) + 2 + 1);
+  sprintf(tmp, "-%c%s", c, arg);
+  return tmp;
+}
+
+static error_t parse_opt(int key, char *arg, struct argp_state *state) {
+  struct cli_args *args = state->input;
+
+  // Handle the run args separately
+  if (args->parse_run) {
```

Comment:
Instead of having a nested for loop in the parse function to parse arguments after a `-run`, `-search`, or others are specified, we set a flag in the parser, and branch to this separate switch statement, which has the same behavior